modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,125 +1,236 @@
 /*
- * PROJECT:         ReactOS kernel-mode tests
- * LICENSE:         LGPLv2+ - See COPYING.LIB in the top level directory
- * PURPOSE:         Kernel-Mode Test Suite Device Interface functions test
- * PROGRAMMER:      Filip Navara <xnavara@volny.cz>
+ * PROJECT:     ReactOS kernel-mode tests
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Test for Device Interface functions
+ * COPYRIGHT:   Copyright 2011 Filip Navara <xnavara@volny.cz>
+ *              Copyright 2021 Mark Jansen <mark.jansen@reactos.org>
+ *              Copyright 2021-2022 Oleg Dubinskiy <oleg.dubinskij2013@yandex.ua>
  */
 
-/* TODO: what's with the prototypes at the top, what's with the if-ed out part? Doesn't process most results */
+/* TODO: Add IoRegisterDeviceInterface testcase */
 
 #include <kmt_test.h>
 #include <poclass.h>
 
 #define NDEBUG
 #include <debug.h>
 
-#if 0
-NTSTATUS
-(NTAPI *IoGetDeviceInterfaces_Func)(
-   IN CONST GUID *InterfaceClassGuid,
-   IN PDEVICE_OBJECT PhysicalDeviceObject OPTIONAL,
-   IN ULONG Flags,
-   OUT PWSTR *SymbolicLinkList);
-
-NTSTATUS NTAPI
-ReactOS_IoGetDeviceInterfaces(
-   IN CONST GUID *InterfaceClassGuid,
-   IN PDEVICE_OBJECT PhysicalDeviceObject OPTIONAL,
-   IN ULONG Flags,
-   OUT PWSTR *SymbolicLinkList);
-#endif /* 0 */
-
-static VOID DeviceInterfaceTest_Func()
+/* Predefined GUIDs are required for IoGetDeviceInterfaceAlias and IoOpenDeviceInterfaceRegistryKey.
+ * Only they can provide the aliases and the needed subkeys, unlike manually declared test GUIDs.
+ * Since IoRegisterDeviceInterface testcase is missing, it is not possible to register the new device interface
+ * and get an alias/key handle of it using this test. */
+/* Invlaid GUID */
+#define STATIC_GUID_NULL \
+    0x00000000L, 0x0000, 0x0000, {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
+/* From https://docs.microsoft.com/en-us/windows-hardware/drivers/stream/ksproperty-topology-categories */
```

Comment:
More like, extracted from our headers? (for example, https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/ks.h;hb=4321c975c76aa149b8b624f360a196eaabf85f54#l103 )

I also wonder why not just using below a construct more similar to this:
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/wdm/audio/sysaudio/main.c;hb=4321c975c76aa149b8b624f360a196eaabf85f54#l19

Reply:
> More like, extracted from our headers? (for example, https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/ks.h;hb=4321c975c76aa149b8b624f360a196eaabf85f54#l103 )

Yeah, it is actually copied from our ks.h header, so I updated the comment accordingly.
> I also wonder why not just using below a construct more similar to this:
https://git.reactos.org/?p=reactos.git;a=blob;f=drivers/wdm/audio/sysaudio/main.c;hb=4321c975c76aa149b8b624f360a196eaabf85f54#l19

Indeed, we can simplify the test GUIDs declarations like it's done there, even without including ks.h here, since it's not available in the kmtests environment, as it was previously discovered by me. So I changed them as suggested.